python/cudnn/sdpa/bwd/api_sm80.py

```
@@ -0,0 +1,651 @@
+
+        # ---- mask token (same resolution as the forward adapter) ------
+        swa_left = self.window_size_left
+        swa_right = self.window_size_right
+        self.right_bound = 0
+        if self.is_causal:
+            self.mask_token = "causal" if swa_left < 0 else "causal_swa"
+            self.swa_window_runtime = max(0, swa_left) if swa_left >= 0 else 0
+            self.right_bound = max(0, swa_right)
+        elif swa_left >= 0 or swa_right >= 0:
+            self._not_implemented_error_if(swa_right > 0, "SM80 BPROP: non-causal SWA with window_size_right > 0 unsupported")
+            self.mask_token = "swa"
```

Ruff (0.16.1)

[error] Possible hardcoded password assigned to: "mask_token"

(S105)